File: apollo-runtime/src/main/java/com/apollographql/apollo/internal/util/Cancelable.java
package com.apollographql.apollo.internal.util;

/**
 * Represents an operation which can be cancelled.
 */
public interface Cancelable {
  /**
   * Cancels the operation.
   */
  void cancel();

  /**
   * Checks if this operation has been cancelled.
   *
   * @return true if this operation has been cancelled else returns false
   */
  boolean isCanceled();
}

File: deprecated-apollo-runtime/src/main/java/com/apollographql/apollo3/internal/util/Cancelable.kt
package com.apollographql.apollo3.internal.util

/**
 * Represents an operation which can be canceled.
 */
interface Cancelable {
  /**
   * Cancels the operation.
   */
  fun cancel()

  /**
   * Checks if this operation has been canceled.
   *
   * @return true if this operation has been canceled else returns false
   */
  val isCanceled: Boolean
}